### PR TITLE
ssh/tailssh: evaluate tailcfg.SSHPolicy on incoming connections

### LINKED ISSUE #3960: bump MapRequest.version before 1.22.0

### PR DESCRIPTION
**Pull request by bradfitz**
Updates #3802
Fixes #3960

### CODE REVIEW

File: ssh/tailssh/tailssh.go [hunk 1]
```diff
@@ -91,35 +121,54 @@ func (srv *server) handleSSH(s ssh.Session) {
...
+	if !isPty {
+		fmt.Fprintf(s, "TODO scp etc\n")
+		s.Exit(1)
+		return
+	}
 	var cmd *exec.Cmd
-	sshUser := s.User()
-	if os.Getuid() != 0 || sshUser == "root" {
+	if os.Getuid() != 0 || localUser == "root" {
 		cmd = exec.Command("/bin/bash")
```
Comment (maisem) [inline ssh/tailssh/tailssh.go:120]: Can we detect the login shell instead of always trying bash?

File: ssh/tailssh/tailssh.go [hunk 2]
```diff
@@ -91,35 +121,54 @@ func (srv *server) handleSSH(s ssh.Session) {
...
+	sctx := &sshContext{
+		now:     time.Now(),
+		sshUser: s.User(),
+		srcIP:   srcIP,
+		node:    node,
+		uprof:   &uprof,
+	}
+	action, localUser, ok := evalSSHPolicy(pol, sctx)
+	if ok && action.Message != "" {
+		io.WriteString(s, action.Message)
```
Comment (maisem) [inline ssh/tailssh/tailssh.go:150]: New line?

File: ssh/tailssh/tailssh.go [hunk 3]
```diff
@@ -66,6 +69,33 @@ type server struct {
 	logf logger.Logf
 }
 
+var debugPolicyFile = envknob.String("TS_DEBUG_SSH_POLICY_FILE")
+
+func (srv *server) sshPolicy() (_ *tailcfg.SSHPolicy, ok bool) {
+	if debugPolicyFile != "" {
+		f, err := os.ReadFile(debugPolicyFile)
```
Comment (maisem) [inline ssh/tailssh/tailssh.go:76]: Not sure how I feel about having a way to bypass domain policy.

File: ssh/tailssh/tailssh.go (hunk 2 above)
Comment (bradfitz) [reply ssh/tailssh/tailssh.go:150]: It's intentional like this to give the server flexibility. The server can include the newline if that's what they want.

File: ssh/tailssh/tailssh.go (hunk 1 above)
Comment (bradfitz) [reply ssh/tailssh/tailssh.go:120]: Sure, but that's unrelated to this PR.

**Comment by bradfitz** [comment]
PTAL

**Review by maisem — APPROVED** [review]

**Pull request merged**